lib/assert.js

```diff
@@ -145,13 +146,7 @@ function _deepEqual(actual, expected, strict) {
   if (actual === expected) {
     return true;
   } else if (actual instanceof Buffer && expected instanceof Buffer) {
-    if (actual.length != expected.length) return false;
-
-    for (var i = 0; i < actual.length; i++) {
-      if (actual[i] !== expected[i]) return false;
-    }
-
-    return true;
+    return bufferBinding.compare(actual, expected) === 0;
```

Comment:
If you will allow me one more nit: if you're using just one function from the buffer binding, you might as well import it directly:

```javascript
const compare = process.binding('buffer').compare;

// ...

return compare(actual, expected) === 0;
```